lib/ip_blocklist_manager.js

```diff
@@ -0,0 +1,88 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * This module is used to manage multiple ip blocklists. It has the
+ * ability to initialize, load and reload itself on any list changes.
+ *
+ */
+var Promise = require('bluebird')
+
+module.exports = function (log, config) {
+  var IPBlocklist = require('./ip_blocklist')(log, config)
+
+  function IPBlocklistManager() {
+    this.ipBlocklists = []
+  }
+
+  IPBlocklistManager.prototype.load = function (lists) {
+    var self = this
+
+    if (!Array.isArray(lists)) {
+      throw Error('lists must be an array')
+    }
+
+    // Initialize and load a blocklist for each file path
+    var loadedLists = lists.map(function (listPath) {
+      var blocklist = new IPBlocklist()
+      self.ipBlocklists.push(blocklist)
+      return blocklist.load(listPath)
+    })
+
+    return Promise.all(loadedLists)
+  }
+
+  IPBlocklistManager.prototype.clear = function () {
+    this.ipBlocklists = []
+  }
+
+  IPBlocklistManager.prototype.contains = function (ipAddress) {
+    var self = this
+    var startTime = Date.now()
+    var endTime
+    var found = false
+
+    // Go through all blocklists and check if there is a match on the
+    // ipAddress.
+    self.ipBlocklists.forEach(function (blocklist) {
```

Comment:
@rfk I can see that being useful. It's not currently in this code, so it seems like unnecessary cpu cycles. To do what you suggest, perhaps an array should be built of list names that contain the address?